### ISSUE TITLE
Make sure that Gateway consumes / forwards all SMS before resuming polling

### ISSUE THREAD

**Issue by SCdF** [medic/medic-gateway#137]
I think right now Gateway only forwards 10 SMS per 60 second poll. This makes sense because we don't want to make a massive request to API, but we should not wait for the next polling cycle. Instead we should loop until we've run out of messages to send.

This should improve performance in cases where Gateway has has trouble waking up for awhile and has a queue of messages to forward.

**Comment by garethbowen** [comment]
Bumped to 3.8 as discussed in the product meeting. Gateway can be released as soon as this is ready.

**Comment by SCdF** [comment]
Ready for AT:
 - Everything should still work (we haven't touched this code base in a long time, and I have never touched it before, so this is a pretty valid test!)
 - Gateway should no longer be gated to sending 10 SMS to Medic per minute (so, if you queued up 100 SMS it won't take 10 requests over 10 minutes, they should come much quicker than that)

**Comment by dianabarsan** [comment]
Assigning myself for AT.

**Comment by dianabarsan** [comment]
Looks good to me! 
I've installed the branch on an Android 5 device, started spamming messages and checking reported_dates of the resulting docs in Couch. There are no 1 minute gaps between batches of messages. 

I did however notice some issues about message statuses that I've detailed here and are most likely not connected to this change, that I'm also seeing on `master` and I've detailed here: https://github.com/medic/medic-gateway/issues/154
